app/main.py

```diff
@@ -1,3 +1,59 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance" or int | float) -> "Distance":
```

Comment:
it is better to use `Union["Distance", int, float]`

Reply:
As far as I remember, It's ok to use  |  operator from Python 3.11.  Maybe "or" doesn't fit enough. Thanks for your review

Reply:
> it is better to use `Union["Distance", int, float]`

As far as I remember, It's ok to use | operator from Python 3.11. Maybe "or" doesn't fit enough. Thanks for your review

Reply:
you are right. it is even recommended in the docs:
> To define a union, use e.g. Union[int, str] or the shorthand int | str. Using that shorthand is recommended.

but using `"Distance" | int | float` will produce an error. to avoid any errors you can wrap this expression in double quotes `"Distance | int | float"`